dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -1,37 +1,106 @@
-{% macro validate_get_incremental_strategy(config) %}
-  {#-- Find and validate the incremental strategy #}
-  {%- set strategy = config.get("incremental_strategy", default="append") -%}
-
-  {% set invalid_strategy_msg -%}
-    Invalid incremental strategy provided: {{ strategy }}
-    Expected one of: 'append', 'delete+insert', 'merge'
-  {%- endset %}
-  {% if strategy not in ['append', 'delete+insert', 'merge'] %}
-    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+{% macro dbt_trino_get_tmp_relation_type(strategy, unique_key, language) %}
+
+  /* {#
+       If we are running multiple statements (DELETE + INSERT),
+       we must first save the model query results as a temporary table
+       in order to guarantee consistent inputs to both statements.
+
+       If we are running a single statement (MERGE or INSERT alone),
+       we can save the model query definition as a view instead,
+       for faster overall incremental processing.
+  #} */
+
+  {% if language == 'sql' and (strategy in ('default', 'append', 'merge') or (unique_key is none)) %}
+    {{ return('view') }}
+  {% else %}  {#--  play it safe -- #}
+    {{ return('table') }}
   {% endif %}
-
-  {% do return(strategy) %}
 {% endmacro %}
 
-{% macro get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
-  {% if strategy == 'append' %}
-    {% do return(get_append_sql(target_relation, tmp_relation, dest_columns)) %}
-  {% elif strategy == 'delete+insert' %}
-    {% do return(get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
-  {% elif strategy == 'merge' %}
-    {% do return(get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns, predicates=none)) %}
+{% materialization incremental, adapter='trino', supported_languages=['sql'] -%}
+
+  {#-- Set vars --#}
+  {%- set full_refresh_mode = (should_full_refresh()) -%}
+  {%- set language = model['language'] -%}
+  {#-- TODO: figure out why we need to add table explicitly, otherwise persist_docs doesn't know the type --#}
+  {% set target_relation = this.incorporate(type='table') %}
```

Comment:
Interestingly the snowflake adapter doesn't have to add `.incorporate(type='table')`, might be a bug in the relation implementation.